### ISSUE TITLE
lose line numbers when using match

### ISSUE THREAD

**Issue by boxxxie** [noprompt/meander#78]
when i get errors in my match dispatches (after a pattern is found) when there are errors in that code i get line numbers in my stack traces pointing to the line of `(match`, not the line where the error occurs.

**Comment by noprompt** [comment]
@boxxxie Thanks for the report. I've run into a little trouble with this myself and, I suspected, it'd only be a matter of time before someone opened an official issue about it.

I haven't given it much thought but, I'm curious, do you have any suggestions? I think there's a trick to this but I can't think of it at the moment.

**Comment by boxxxie** [comment]
the `match` macro looks like it evaluates more like a function. when i macro expand i get the same code i typed.

**Comment by noprompt** [comment]
@boxxxie Can you share an example with me? The example below seems to be reporting the correct line numbers.

```clj
(defn error []             ;; Line 4
  (throw (ex-info "" {}))) ;; Line 5

(m/match 1                 ;; Line 7
  1
  (error))                 ;; Line 9
;; =>
;; 1. Unhandled clojure.lang.ExceptionInfo
;;
;; {}
;; REPL:    5  scratch/error
;; REPL:    4  scratch/error
;; REPL:    9  scratch/eval14317
;; REPL:    7  scratch/eval14317
```

**Comment by jimmyhmiller** [comment]
I think I understand what is meant here. When you have an error that doesn't match, we don't tell you what doesn't match and what line it is on. For example:

```clojure
(m/match {:x {:y {:z 2}}} ;; line 980
  {:x
   {:y
    {:z 1}}}
  :match)
```
Throws this exception

```clojure
1. Unhandled clojure.lang.ExceptionInfo
   non exhaustive pattern match
   {}
                      REPL:  980  wander.core10/eval17978/FAIL
                      REPL:  980  wander.core10/eval17978
                      REPL:  980  wander.core10/eval17978
                     ...
```

It would be great if we could instead say something like this:

```
980| (m/match {:x {:y {:z 2}}}
981|   {:x
982|    {:y
983|     {:z 1}}}
             ^
984|   :match)

On line 983 above, the pattern was looking to match 1, but instead we found 2.
```

Or if we had a pred instead

```
980| (m/match {:x {:y {:z 1}}}
981|   {:x
982|    {:y
983|     {:z (m/pred string?)}}}
             ^^^^^^^^^^^^^^^^
984|   :match)

On line 983 above, the pattern was looking for a string, but instead we found 1 which is a number.
```

We could be even better by showing where in the data structure that is, but that will definitely require some heuristics. 

Basically, we could have elm style error messages.

I think this is definitely possible, but will require a lot of work.

**Comment by noprompt** [comment]
@jimmyhmiller Maybe @boxxxie can clear it up as I took

> when i get errors in my match dispatches (after a pattern is found)

as errors that are thrown on the right side of the match, not errors due to non-exhaustion.

**Comment by noprompt** [comment]
I'm going to close this due to inactivity.
